EIPS/EIPs/eip-5075.md

```diff
@@ -0,0 +1,294 @@
+        }
+    }
+}
+```
+**Note**
+
+The actual limit would be based on the previous 1h outflows and should that be 0 the limit approaches 2*rateLimit for that 1h slot, though to be noted does not apply for following slots so rather than 2x, net outflow limit for any time t can be seen as
+
+rateLimit available at time 0 + timeWindow  * rateLimit
+
+Not simply timeWindow * rateLimit, where 2 * rateLimit for cases where the start outflow (at time 0 for the window) would be 0 for the previous 1h (no activity last 1h), as the outflow at time 0 increases the excess possible decreases.
+
+So let's say that you have a timeWindow at 100m and a rateLimit at 100 bips (lets say theres 1000 tokens and so 100 tokens per 100m)
+```
+Time (m)
+Amount out (up to limit)
+000 005 010 020 040 060 080 099 100 105 110 120...
+100 005 010 010 020 020 020 014 001 005 005 010...
+Delta 1h
+100+005+005+010+020+020+020+019+001 (200 last 1h possible at 0-100m passed)
+....005+005+010+020+020+020+019+001+005 (105 last 1h possible at 05-105m passed)
+
+Amount out last 1h window
+100 105 115 125 145 165 185 199 200 105 105 110 ...
+```
+
+So realistically ends up as the excess to the rateLimit possible being the free rate available at time 0 for the window.
+
+**Note**
+
+Making the timeWindow and rateLimit small for that window limits available rate at t0. So timeWindow at 5m, rateLimit 5 may be better than 100m and 100 bips.
+
+Though having a a large allowance and window allow spike capacity that smoothes and makes the av rate approach target rather than a set absolute target. could be useful. At worst with a 100 bip, 1h window, would take an attacker 9h to extract vs 10 as t0 would be 1h worth excess, but they can only do up to the rateLimit every 1h after as the previous 1h would have reached the rateLimit and not be 0.
+
+
+## Security Considerations
+The above discussion shows why we decided to implement the per token tracking and why and how this should be used to help limit losses from hacks. Contracts must consider any whitelists very carefully as these may be able to be used to drain assets by themselves having been hacked or through non intended use. Teams and projects must also consider their usual outflows based on analytics to try and not impact user experience, balancing both the limit in potential damage from the hack as well as the user experience from transaction reverts. The getLimitLeft can be used in the front end to let users know that there is high demand and to wait for outflows from the contracts. This also does not affect approvals and transferFrom to be used on the contracts balances, so contracts must be aware and manage approvals carefully as these are not checked, using transferL where possible.
+  
+Note rateLimit can be set to 1000 (100% in basis points) to allow for the feature to be effectively disabled. Had considered doing a flag and a check but this would create a slight gas inefficiency on all transactions during usual operation.
```

Comment:
```suggestion
Note that `rateLimit` may be set to 1000 (100% in basis points), which effectively disables the rate limit. A flag had been considered, but it would result in higher gas fees and therefore was deemed unneccesary.
```